tests/material/bingham_test.cc

```diff
@@ -0,0 +1,565 @@
+    REQUIRE(material->property("density") ==
+            Approx(jmaterial["density"]).epsilon(Tolerance));
+  }
+
+  SECTION("Bingham check stresses with no strain rate") {
+    unsigned id = 0;
+    auto material = Factory<mpm::Material<Dim>, unsigned>::instance()->create(
+        "Bingham2D", std::move(id));
+    REQUIRE(material->id() == 0);
+
+    // Initialise material
+    Json jmaterial;
+    jmaterial["density"] = 1000.;
+    jmaterial["youngs_modulus"] = 1.0E+7;
+    jmaterial["poisson_ratio"] = 0.3;
+    jmaterial["tau0"] = 771.8;
+    jmaterial["mu"] = 0.0451;
+    jmaterial["critical_shear_rate"] = 0.2;
+
+    material->properties(jmaterial);
+
+    // Check material status after assigning material property
+    REQUIRE(material->property_handle() == true);
+
+    // Initialise stress
+    mpm::Material<Dim>::Vector6d stress;
+    stress.setZero();
+    REQUIRE(stress(0) == Approx(0.).epsilon(Tolerance));
+    REQUIRE(stress(1) == Approx(0.).epsilon(Tolerance));
+    REQUIRE(stress(2) == Approx(0.).epsilon(Tolerance));
+    REQUIRE(stress(3) == Approx(0.).epsilon(Tolerance));
+    REQUIRE(stress(4) == Approx(0.).epsilon(Tolerance));
+    REQUIRE(stress(5) == Approx(0.).epsilon(Tolerance));
+
+    // Initialise dstrain
+    mpm::Material<Dim>::Vector6d dstrain;
+    dstrain.setZero();
+    dstrain(0) = 0.0010000;
+    dstrain(1) = 0.0005000;
+    dstrain(2) = 0.0000000;
+    dstrain(3) = 0.0000000;
+    dstrain(4) = 0.0000000;
+    dstrain(5) = 0.0000000;
+
+    // Add particle
+    mpm::Index pid = 0;
+    Eigen::Matrix<double, Dim, 1> coords;
+    coords << 0.5, 0.5;
+    auto particle = std::make_shared<mpm::Particle<Dim, 1>>(pid, coords);
+
+    // Coordinates of nodes for the cell
+    mpm::Index cell_id = 0;
+    const unsigned Dim = 2;
+    const unsigned Dof = 2;
+    const unsigned Nphases = 1;
+    const unsigned Nnodes = 4;
+    const unsigned phase = 0;
+    const double dt = 1;
+
+    coords << -2, -2;
+    std::shared_ptr<mpm::NodeBase<Dim>> node0 =
+        std::make_shared<mpm::Node<Dim, Dof, Nphases>>(0, coords);
+    coords << 2, -2;
+    std::shared_ptr<mpm::NodeBase<Dim>> node1 =
+        std::make_shared<mpm::Node<Dim, Dof, Nphases>>(1, coords);
+    coords << 2, 2;
+    std::shared_ptr<mpm::NodeBase<Dim>> node2 =
+        std::make_shared<mpm::Node<Dim, Dof, Nphases>>(2, coords);
+    coords << -2, 2;
+    std::shared_ptr<mpm::NodeBase<Dim>> node3 =
+        std::make_shared<mpm::Node<Dim, Dof, Nphases>>(3, coords);
+
+    std::shared_ptr<mpm::ShapeFn<Dim>> shapefn =
+        Factory<mpm::ShapeFn<Dim>>::instance()->create("SFQ4");
+
+    auto cell = std::make_shared<mpm::Cell<Dim>>(cell_id, Nnodes, shapefn);
+
+    cell->add_node(0, node0);
+    cell->add_node(1, node1);
+    cell->add_node(2, node2);
+    cell->add_node(3, node3);
+
+    particle->assign_cell(cell);
+    particle->compute_shapefn();
+    particle->compute_strain(phase, dt);
+
+    // Reset stress
+    stress.setZero();
```

Comment:
Remove this line

Reply:
I think we need it to make sure the stress we are inputting is all 0, instead of the stress from previous case.